File: 1347-minimum-number-of-steps-to-make-two-strings-anagram/1347-minimum-number-of-steps-to-make-two-strings-anagram.java
class Solution {
    public int minSteps(String s, String t) {
        int[] sMap = new int[26];
        int[] tMap = new int[26];
        
        map(s, sMap);
        map(t, tMap);
        int count = 0;
        for(int i = 0; i < 26; ++i){
            if(tMap[i] != sMap[i]){
                count += Math.abs(tMap[i] - sMap[i]);
            }
        }
        
        return count/2;
    }
    
    public void map(String s, int[] map){
        for(int i = 0; i < s.length(); ++i){
            map[s.charAt(i) - 'a']++;
        }
    }
}

File: 1347-minimum-number-of-steps-to-make-two-strings-anagram/1347-minimum-number-of-steps-to-make-two-strings-anagram.py
class Solution:
    def minSteps(self, s: str, t: str) -> int:
        s_map = self.map_s(s)
        t_map = self.map_s(t)
        s_diff, t_diff = 0,0
        
        for i in range(26):
            
            if t_map[i] < s_map[i]:
                t_diff += s_map[i] - t_map[i]
       
        
        return t_diff
    
    def map_s(self, s):
        c_map = [0] * 26
        
        for c in s:
            idx = ord(c) - ord('a')
            c_map[idx] += 1
        
        return c_map